src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -406,7 +406,11 @@ private static function createRedirectResolver(array $options, string $host, ?ar
                     $redirectHeaders['no_auth'] = array_filter($redirectHeaders['no_auth'], $filterContentHeaders);
                     $redirectHeaders['with_auth'] = array_filter($redirectHeaders['with_auth'], $filterContentHeaders);
 
-                    stream_context_set_option($context, ['http' => $options]);
+                    if (\PHP_VERSION_ID >= 80300) {
+                        stream_context_set_options($context, ['http' => $options]);
```

Comment:
shouldn't we define a polyfill for `stream_context_set_options` in `symfony/polyfill-php83` and then always use it ?

Reply:
there's one already but I thought this was simple enough to fix without the polyfill